packages/handlersjs-core/lib/storage/memory-store.spec.ts

```diff
@@ -0,0 +1,129 @@
+
+import { MemoryStore } from './memory-store';
+
+describe('MemoryStore', () => {
+
+  interface TestInterface {
+    key1: number;
+    key2: string;
+    key3: boolean;
+    key4: string[];
+  }
+
+  let memoryStore: MemoryStore<TestInterface>;
+
+  beforeEach(() => {
+
+    memoryStore = new MemoryStore();
+
+  });
+
+  it('should not iterate over keys pairs when it is empty', async () => {
+
+    const elements = [];
+    for await(const entry of memoryStore.entries()) elements.push(entry);
+
+    expect(elements.length).toBe(0);
+
+  });
+
+  it('should not have keys that were not added', async () => {
+
+    expect(await memoryStore.has('key1')).toBeFalsy();
+    expect(await memoryStore.get('key1')).toBeUndefined();
+    expect(await memoryStore.latestUpdate('key1')).toBeUndefined();
+    expect(await memoryStore.hasUpdate('key1', Date.now())).toBeUndefined();
+
+  });
+
+  it('should get a value that has been set', async () => {
+
+    await memoryStore.set('key1', 5);
+
+    expect(await memoryStore.get('key1')).toEqual(5);
+
+    await memoryStore.set('key2', 'test');
+    await memoryStore.set('key3', true);
+    await memoryStore.set('key4', [ '123', '321' ]);
+
+    expect(await memoryStore.get('key2')).toEqual('test');
+    expect(await memoryStore.get('key3')).toEqual(true);
+    expect(await memoryStore.get('key4')).toEqual([ '123', '321' ]);
```

Comment:
These don't really test anything more, since at runtime the types do not matter.